### DISCUSSION TITLE
CPU spikes at 96K session

### DISCUSSION THREAD

**Discussion by fojerhar** [apohl79/audiogridder#1004] [General]
Hi

Usually I have to work with 96K sessions in protools.
At 48K I can use a ton of heavy plugins for mixing, and at protools system usage I can see that the processing is spreaded nicely between the 64 cores.
But in 96K the two mid cores are overused and pt stops with 9173 overload error.
I tried to increase the AG buffer. at 48K I use 2 blocks, but even 4,8 doesn't help. 12 and above would be good, but in this case PT cannot compensate the 48000 (or above) sample latency, so it is not useable.
Now I converted a 96K mix session into 48K and the difference can be seen really well. Same sessions, same plugins, only the 4 blocks on 96K and 2 blocks on 48K is the difference
Here are the screenshots:
![Screen Shot 2022-08-31 at 18 48 47](https://user-images.githubusercontent.com/72139943/187738735-58bfbe7a-02cd-4ff3-bca3-a13198763bd8.png)
![Screen Shot 2022-08-31 at 19 02 32](https://user-images.githubusercontent.com/72139943/187738757-1120e87c-f9f0-4ffa-809f-c4c46a561e8d.png)

Am I doing something wrong? Or how should I solve this 96K issue.
(by my ears I believe in 96K, depth and dimension sounds much better, even if the end it is converted down to a 320kbps mp3)

thank you
Viktor

**Comment by apohl79** [comment]
Hey Viktor,

I don't have an answer right now, but I will start some experiments and see if things can be optimized. Could you share some logs of the FX plugin?

Edit: In case you have tracing enabled in the plugin or the server, can you disable it (it causes some overhead)?

Cheers,
Andreas

**Comment by fojerhar** [comment]
Hi Andreas
Thank you, here are the log zip
[FX.zip](https://github.com/apohl79/audiogridder/files/9471513/FX.zip)

At the beginning I opened the first 96K session, I had a lot of freezing at PT, you will see in the logs.Changed the buffer from 2 to 8 blocks also.
After that I opened the same converted session to 48K, what works well.
After that I reopened the first 96K session, where I deleted some plugins (Acustica El Rey) and the spikes came lower and swapped the standard acustica plugins to the ZL (zero latency version, what in normal case uses more cpu that the standard one, but it works better for some reason with AG).

So I have found some "buggy" plugins, what cause the peaks, but that comes only at 96K, at 48K there is no issue with it. If you are familiar with this company, this El Rey has a bit older engine (core16).

I know that that sample rate doubles up, more data needs to be real time processed, what needs more cpu power.
I also accept that because all the tracks are summed in the master bus, the processing cannot be truly parallel because by the delay compensation and processing order the computing has to be serial, each after each, because plugins setted up in serial.

I don't exactly know what shows the protools system usage. 
I don't know if it is true or not.Sometimes it stays at 2-3 percent while AG servers works with 100 plugins at background. I don't know, how it shows the multicore threadings, or if it knows that something is also happening outside the daw.

Thank you for your smart and hard work again and again. This program helps a lot for staying creative, and deal less with technical limitations.

Btw: if I save the session with beta12, and reinstall beta7 (what is the most stable for me) all the AG plugins are empty in protools, but luckily beta12 seems fine for me, if something collapses on the sandbox chain it solves it fast and smart.

I use chain isolation, tried the plugin isolation also, but did not help
Tracing was already off, but now is on, and do not cause problems.

I had this 96K spike issue from the beginning. I record all the albums at 96K and where I reach the limitations I convert the whole session to 48K and continue and finish the mix there.

For being clear: my computer is a hackintosh with os 12.5.1 Monterey with ADM Ryzen threadripper 3970x cpu
If you need any special local test on my computer, just let me know

bests
Viktor

**Reply by apohl79** [reply]
Hey Viktor,

first of all, going from 48k to 96k reduces the available realtime processing time of a chunk of samples by half. For example, if the block size is 1024 samples, the available time is ~20ms for all plugins inserted on a track in PT but with 96k it is only ~10ms. So the realtime pressure doubles.

I did some testing and added more stats to AG and the tray monitor app to get a better sense of whats going on and what plugin chain might cause an issue. I will also added much stricter timeouts to the audio streamer, so it should prevent the DAW from stopping in most cases by dropping audio internally.

The problem is when the read buffer queue of a plugin runs empty, and AG has to block to wait for audio coming back from the server. This should only happen (I'm talking about local mode now, as over the network there might be other factors as well) if a plugin chain is constantly using more time then available (~10ms with 96k). 

In my tests I've used 32 tracks with one AG plugin on each. With a 4 blocks buffer everything is stable with a latency (that you can see in the monitor) or around 1-1.5ms. Then I added a view acustica plugins step by step to get closer to the 10ms (I had to change to 8 blocks). As long as I stayed below 10ms, things were fine. Approaching 10ms and above leads to the queue running empty, AG blocking and PT erroring. 

So, check the monitor and take a closer look at the individual chain latencies. The next AG version might help a bit more by showing read queue levels and read errors and will try to avoid PT hangs (you would still have audio drop outs). It looks like you might be hitting a wall, as the combination of plugins you are using might not be fast enough to process the audio in realtime.

> I don't exactly know what shows the protools system usage.
I don't know if it is true or not.Sometimes it stays at 2-3 percent while AG servers works with 100 plugins at background. I don't know, how it shows the multicore threadings, or if it knows that something is also happening outside the daw.

It looks like PT measures the time an audio thread is active (runs calculations and does not wait) and the time it needs to process sample buffers. As the AG plugin is not doing audio computations, only the response time matters. The longer it has to wait for data, the higher the meter in PT. That means, if you use buffering, and the queues don't run empty, the system usage meter in PT should stay very low, as the AG plugins can return immediately.

> Btw: if I save the session with beta12, and reinstall beta7 (what is the most stable for me) all the AG plugins are empty in protools, but luckily beta12 seems fine for me, if something collapses on the sandbox chain it solves it fast and smart.

Thx, will have a look.

> Tracing was already off, but now is on, and do not cause problems.

It will make things slightly slower, so I suggest to keep it off until you have to report a crash bug, thats where it can be very helpful for me.

Andreas

**Reply by fojerhar** [reply]
Hi Andreas

Thank you for the answer.
Logical and make sense. With this approach (to keep chains under 10ms) helps solving issues. With this, I just split up the chan, and using 3-4 AG plugs in serial and you was right, no issues or cpu spikes (just some collapses, but that is fine).

For this case a feature would be nice, if we could drag and drop (even copy) plugins between AG instances. In this case I save the channel preset, delete the last 4 plugins, load a new AG load the preset and deleting the first 3 instances, and now I spliced up the chain.

The other issue. At home. I use the beta12. In the studio I wanted to continue the session, there we have the M1studio ultra. On the M1 beta12 has some buffering issues. If I put AG on a channel that is fine, if I put it on a bus there is a "robotic" noise, when the buffering has issues, so cannot use it, but beta9 doesn't open the beta12 saved sessions, so could not continue. Long solution save all instances into presets in b12 and load the at b9. (I see I already mentioned it, sorry :) )
Just let you know this issue.

**Anyhow! The most important thing** is that without your AG I could not be able to make of my mixes with this quality, what I can do now, huge thank you for that, thank yo for the smart work, you are doing a better job than big companies does.....

**Reply by apohl79** [reply]
understood! yes drag&drop support is already in the list of feature requests. cheers!

**Comment by Henri75** [comment]
It looks a lot like some issues I got in 96k too (but I stopped trying to use AG in 96k).

In 48k I have less problem, but still run into some issues (see my other posts). Lately on a session with a dozen AG insert, remote server under 10% usage, 10% cpu bandwith, session is running fine in S1 until I activate AG inserts then I got dropout in the session.

I have excluded plugin issues or ressources after a lot of tests. Issues seems to be more related to the accumlation of AG inserts, whatever are the plugins used on the remote server. Again it happens on large projects. A few AG inserts will always works. At 96k things are getting worst faster.

Not sure how I can help, but I will be glad to !

**Comment by Henri75** [comment]
By the way I use a lot of AA plugins, and some are known to have issues (cpu spikes) even when used out of AG (for example in Studio One or Cubase). Some works perfectly, and indeed ZL versions are more predictable in cpu suage and latency.

**Comment by Henri75** [comment]
Great ! Would be really helpful !

A copy/paste buttons for plugin/or whole chain between plugins would be
helpful too, sometimes easier than drag and drop

Le lun. 5 sept. 2022 à 09:10, Andreas Pohl ***@***.***> a
écrit :

**Reply by apohl79** [reply]
Got it, added it to the FR ticket

**Comment by Henri75** [comment]
Thanks, Drag/Drop and copy/paste will be so helpful for the workflow and
will make AG use easier and faster !

Le lun. 5 sept. 2022 à 11:51, Andreas Pohl ***@***.***> a
écrit :

**Comment by fojerhar** [comment]
Hi Andreas
This issue, to keep the chains under 10ms concept is working, that I spliced up my chains, and using different buffers on tracks where more is needed, and I can do the mix at 96K, big thank you, an thank you for the proper answers also

**Reply by apohl79** [reply]
awesome!
